### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by arpcorrea**
Hello,
I would like to merge my dev code with your master. Here, as discussed, only TwoPlates.py is hard coded. RectangularDomain objects are now simpler, containing only masks, name, position (this last one I still need to work over it... couldn't find a solution yet) and local_fields. Mesher reads the local_fields of each RectangularDomain objects and sum them, creating Fields (like "global fields from locals").

### CODE REVIEW

**Review by ilyasst — CHANGES_REQUESTED** [review]
Please check my comments.

File: consolidate/problems/domains.py [hunk 1]
```diff
@@ -37,90 +36,102 @@ def generate_mask(self, totalNy, totalNx):
                     self.mask_external_boundary.update({"Left Edge":mask_left})
                     self.mask_external_boundary.update({"Right Edge":mask_right})
                     
-                    if self.position ==1:
-                        mask_bottom[self.elements["Elements in Y"][0]][y_i+1]=1
-                        contact_mask[self.elements["Elements in Y"][1]][y_i] = 1
+                    if position ==1:
+                        mask_bottom[self.elementsY[0]][y_i+1]=1
+                        contact_mask[self.elementsY[1]][y_i] = 1
                         self.mask_external_boundary.update({"Bottom Edge": mask_bottom})
```
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/problems/domains.py:42]: true... need to think about that...

File: consolidate/problems/domains.py [hunk 2]
```diff
@@ -37,90 +36,102 @@ def generate_mask(self, totalNy, totalNx):
...
                         self.mask_contact_interface.update({"Top Edge": contact_mask})
```
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/problems/domains.py:43]: true... need to think about that...

**Comment by arpcorrea** [comment]
Please have a look in the updates of this PR.

**Comment by ilyasst** [comment]
Hey @arpcorrea what's up ? :)
Was wondering about this PR, should it still be open ? Where is the latest version right now ?
Should I review this one ?
Thanks

**Comment by arpcorrea** [comment]
yes, it is still opened... it is the newest version (before overcoming the numerical issues)

**Comment by ilyasst** [comment]
Should I review it ? If you have a latest version, could you update it please ?

Ilyass

**Comment by arpcorrea** [comment]
Yes please, go ahead with the review. I still do not have any further update yet (I am still looking around what to do to overcome the numerical issue - we talk about it on Wednesday). Remembering that this version can be ANY PLATE and it will automatically determine the borders (left, right - and top/bottom for the domains in the ends) and contacts (top and/or bottom for the inner domains). The unique condition for the code run properly is to define in the yaml file the domains as:
 "Something X", "Something X+1", ... "Something Xn". In practical example, it follows: "Plate 1", "Plate 2", "Plate 3" or "Domain 1", "Domain 2", "Domain 3" or even "Adherent 1", "Adherent 2", "Adherent 3".

File: consolidate/problems/twoplates.py [hunk 3]
```diff
@@ -1,88 +1,203 @@
...
         for domain in self.domains:
-            domain.set_IC(domain.name, deck)
\ No newline at end of file
+            self.mask_external_boundary = {}
+            deck_dir = deck.doc["Domains"][domain.name]
+            bc_ext = {}
+            for edge in deck_dir["Boundary Condition"]["External"]:
+                bc_ext[edge] = np.zeros((self.totalNy+2, self.totalNx+2))
+
+    def populate_fields_locally(self,deck):
```
Comment (ilyasst) [inline consolidate/problems/twoplates.py:129]: I am currently thinking that this functions should not be part of of twoplates. 
Can the content of this function can be applied to any other domain ?

**Review by ilyasst — CHANGES_REQUESTED** [review]

File: consolidate/models/heat_transfer.py [hunk 4]
```diff
@@ -1,22 +1,60 @@
...
-        self.dy2 = meshes.dy*meshes.dy
-        self.rho=meshes.RhoTotal
-        self.cp=meshes.CpTotal
-        self.k=meshes.KtotalX
+    def __init__(self, mesh, problem):
+        fields ={}
+        for aux in mesh.fields:
+            fields[aux.name]= aux.value
+        self.dt = float(problem.SimulationParameters["Step Time"])
+        self.dx = fields["dx"]
```
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/models/heat_transfer.py:10]: oh yeah... much smarter

[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/models/heat_transfer.py:10]: actually it doesn't work... each field is an object, with name and value...

[the remark this answers is not in the document]
Comment (ilyasst) [reply consolidate/models/heat_transfer.py:10]: You should still be able to do this, even if each field is an object. I probably got this wrong here `self.fields.dx`, but it should work.

File: consolidate/solver/solves.py [hunk 5]
```diff
@@ -5,31 +5,33 @@
 
 class SolvesTwoPlates:
```
Comment (ilyasst) [inline consolidate/solver/solves.py:6]: Cool name !

**Review by ilyasst — CHANGES_REQUESTED** [review]
Please remove the comments and various pdbs (especially in the solver).
Also, is filename.csv necessary ?

I will merge after that.

File: consolidate/problems/domains.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/problems/domains.py]: done, solve

File: consolidate/problems/domains.py [hunk 6]
```diff
@@ -37,90 +36,102 @@ def generate_mask(self, totalNy, totalNx):
...
-                        contact_mask[self.elements["Elements in Y"][0]][y_i] = 1
+                        contact_mask[self.elementsY[0]][y_i] = 1
                         self.mask_contact_interface.update({"Bottom Edge": contact_mask})
-                    if self.position ==2:
-                        contact_mask_middle_bottom[self.elements["Elements in Y"][0]][y_i] = 1
-                        contact_mask_middle_top[self.elements["Elements in Y"][1]][y_i] = 1
+                    if position ==2:
+                        contact_mask_middle_bottom[self.elementsY[0]][y_i] = 1
+                        contact_mask_middle_top[self.elementsY[1]][y_i] = 1
                         self.mask_contact_interface.update({"Bottom Edge": contact_mask_middle_bottom})
```
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/problems/domains.py]: done, solved

File: consolidate/mesher/mesh.py [hunk 7]
```diff
@@ -1,22 +1,7 @@
-import numpy as np
+# import numpy as np
 
-class Mesh:
+# class Mesh:
```
[the remark this answers is not in the document]
Comment (arpcorrea) [reply consolidate/mesher/mesh.py]: new approach... local meshes are created in problem for later merge into a master mesh, done by the "Mesh" class

**Pull request merged**